### PR TITLE
docs: Reorganize README for better understanding

### PR DESCRIPTION
**Pull request by Xuanwo**
Signed-off-by: Xuanwo <github@xuanwo.io>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,11 +1,69 @@
-# storage
+# go-storage
 
 [![Build Status](https://github.com/beyondstorage/go-storage/workflows/Unit%20Test/badge.svg?branch=master)](https://github.com/beyondstorage/go-storage/actions?query=workflow%3A%22Unit+Test%22)
 [![Go dev](https://pkg.go.dev/badge/github.com/beyondstorage/go-storage/v4)](https://pkg.go.dev/github.com/beyondstorage/go-storage/v4)
 [![License](https://img.shields.io/badge/license-apache%20v2-blue.svg)](https://github.com/beyondstorage/go-storage/blob/master/LICENSE)
 [![go storage dev](https://img.shields.io/matrix/go-storage:aos.dev.svg?server_fqdn=chat.aos.dev&label=%23go-storage%3Aaos.dev&logo=matrix)](https://matrix.to/#/#go-storage:aos.dev) <!-- Need update after matrix updated -->
 
-A storage abstraction beyond the existing storage services.
+Storage abstraction that focus on neutral cross-cloud data operation.
```
Comment (xxchan) [inline README.md:8]: What about "vendor-neutral storage abstraction that focuses on cross-cloud data operation."?

Btw, is "cross-cloud" accurate?

Comment (Xuanwo) [reply README.md:8]: Let me think about it.

File: README.md [hunk 2]
```diff
@@ -26,100 +86,215 @@ A storage abstraction beyond the existing storage services.
...
+- [zip](https://github.com/beyondstorage/go-service-zip): zip files
 
-- List: list all Storager in service
-- Get: get a Storager via name
-- Create: create a Storager
-- Delete: delete a Storager
+More service ideas could be found at [Service Integration Tracking](https://github.com/beyondstorage/go-storage/issues/536).
 
-### Storager operation support
+### Complete and easily expandable interface
```
Comment (xxchan) [inline README.md:110]: extensible? (extensibility)

File: README.md [hunk 3]
```diff
@@ -1,11 +1,69 @@
...
+    }
+
+    // Delete hello.txt
+    err = store.Delete("hello.txt")
+}
+```
+
+More examples could be found at [go-storage-example](https://github.com/beyondstorage/go-storage-example).
 
 ## Goal
```
Comment (xxchan) [inline README.md:68]: What about putting Goal at the beginning to attract users?

Comment (xxchan) [reply README.md:68]: And what about adding the slogan "write once, run on every storage service"

File: README.md [hunk 4]
```diff
@@ -26,100 +86,215 @@ A storage abstraction beyond the existing storage services.
...
+_ = om.ServerSideEncryptionCustomerAlgorithm // this object's sse algorithm
+```
 
-func main() {
-	// Init a service. 
-	store, err := services.NewStoragerFromString("fs:///tmp")
-	if err != nil {
-		log.Fatalf("service init failed: %v", err)
-	}
+### Strong Type Everywhere
```
Comment (xxchan) [inline README.md:244]: should be "strong typing" or "strongly typed"

File: README.md [hunk 5]
```diff
@@ -1,11 +1,69 @@
-# storage
+# go-storage
 
 [![Build Status](https://github.com/beyondstorage/go-storage/workflows/Unit%20Test/badge.svg?branch=master)](https://github.com/beyondstorage/go-storage/actions?query=workflow%3A%22Unit+Test%22)
 [![Go dev](https://pkg.go.dev/badge/github.com/beyondstorage/go-storage/v4)](https://pkg.go.dev/github.com/beyondstorage/go-storage/v4)
 [![License](https://img.shields.io/badge/license-apache%20v2-blue.svg)](https://github.com/beyondstorage/go-storage/blob/master/LICENSE)
 [![go storage dev](https://img.shields.io/matrix/go-storage:aos.dev.svg?server_fqdn=chat.aos.dev&label=%23go-storage%3Aaos.dev&logo=matrix)](https://matrix.to/#/#go-storage:aos.dev) <!-- Need update after matrix updated -->
```
Comment (xxchan) [inline README.md:6]: What about adding site and doc and communication like tikv?
image

File: README.md [hunk 6]
```diff
@@ -26,100 +86,215 @@ A storage abstraction beyond the existing storage services.
...
-- Write: write content into file
-- Stat: get file's metadata
-- Delete: delete a file or directory
-- List: list file in prefix or dir styles
+- Metadata: get `Storager` metadata
+```go
+meta := store.Metadata()
+_ := meta.GetWorkDir() // Get object WorkDir
+_, ok := meta.GetWriteSizeMaximum() // Get the maximum size for write operation
+```
```
Comment (xxchan) [inline README.md:119]: Will the code snippets too wordy? (I'm OK with it though)

Comment (xxchan) [reply README.md:119]: Btw, whether adding code snippets here, I think we need to add them to the doc later. 

And I'm wondering the relationship between this README and doc homepage https://beyondstorage.io/docs/go-storage/index

Comment (Xuanwo) [reply README.md:119]: I think the doc homepage should be redesigned.

File: README.md (hunk 1 above)
Comment (Xuanwo) [reply README.md]: I simplify the description to

A **vendor neutral** storage library for Golang.

Any ideas?

Comment (xxchan) [reply README.md]: But I think the hyphen is needed (vendor-neutral)?

**Review by xxchan — APPROVED** [review]
